### PR TITLE
Mm mesh 1761 uplift client to mesh v2 api

### PR DESCRIPTION
**Pull request by matt-mercer**

### CODE REVIEW

File: CHANGE-LOG.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- `message.status` are also lower case `accepted`, `acknowledged`, `undeliverable` rather than `Accepted`, `Acknowledged`. 
+
+##### - Default Retries
+- previously `send_chunk` retries could be optionally enabled by setting the `max_chunk_retries` when creating the client object this has been replaced with the `MeshRetry` based on the standard `urlib3.Retry` object in the HTTPAdapter
+- By default, requests will have two retries with an exponential backoff of `500ms`, but initial `send_message` requests will not be retried.
+- If you wish to disable retries completely set `max_retries=0`
+- Timings and specific behaviour can be tuned using `retry_backoff_factor`, `retry_status_force_list` and `retry_methods` or completely overriden by passing in `max_retries=Retry(...)`
+
+#### - ENDPOINT changes
+- HSCN endpoints are 'deprecated', and internet facing endpoints should be used in preference for clarity HSCN endpoints have been renamed, prefixed `DEPRECATED_HSCN_`.
```
Comment (davidhallam4-nhs) [inline CHANGE-LOG.md]: `preference. For clarity`

File: README.md [hunk 2]
```diff
@@ -14,14 +19,14 @@ Example use
...
-with MeshClient(
-          NHS_DEP_ENDPOINT,
-          'MYMAILBOX',
-          'Password',
-          cert=('/etc/certs/cert.pem', '/etc/certs/key.pem')  # Mesh uses SSL, so you'll need some certs
-        ) as client:
+from mesh_client import MeshClient, DEPRECATED_HSCN_DEP_ENDPOINT
 
+with MeshClient(
+        DEPRECATED_HSCN_DEP_ENDPOINT,
```
Comment (davidhallam4-nhs) [inline README.md]: Do you want to use `INT_ENDPOINT` here?

File: mesh_client/__init__.py [hunk 3]
```diff
@@ -318,7 +411,8 @@ def ping(self) -> dict:
 
     def handshake(self):
```
Comment (davidhallam4-nhs) [inline mesh_client/__init__.py:412]: no return type - same for a bunch of other methods - well it was. There's still iterate_all_messages() and iterate_messages(). I've just had to refresh so I think you may have just added some return types :-)

File: mesh_client/__init__.py [hunk 4]
```diff
@@ -568,14 +662,78 @@ def acknowledge_message(self, message_id: str):
...
+            workflow_filter (Optional[str]): workflow filter string
+
+        Returns:
+            Generator[str]: message ids
+        """
+
+        params: Dict[str, Union[int, str]] = {}
+        if batch_size:
+            if batch_size < 10:
+                raise ValueError("if set page_size should be >= 10")
```
Comment (davidhallam4-nhs) [inline mesh_client/__init__.py]: batch_size

**Review by davidhallam4-nhs — APPROVED** [review]

File: mesh_client/__init__.py (hunk 4 above)
Comment (matt-mercer) [reply mesh_client/__init__.py]: agreed

File: mesh_client/__init__.py (hunk 3 above)
Comment (matt-mercer) [reply mesh_client/__init__.py:412]: handshake not expected to return anything, so think this is correct

File: README.md (hunk 2 above)
Comment (matt-mercer) [reply README.md]: good shout

File: CHANGE-LOG.md (hunk 1 above)
Comment (matt-mercer) [reply CHANGE-LOG.md]: agreed

**Pull request merged**